src/events/ready.ts

```diff
@@ -9,34 +9,53 @@ export default class ReadyEvent {
   }
 
   async run() {
-    if (this.client.user.id === '708014856711962654' && this.client.shard.ids[0] === 0) {
-      const DiscordBotList = new Dbl(this.client);
-      DiscordBotList.init();
-      setInterval(async () => {
-        const atividade = await http.getActivity(this.client.shard.ids[0]);
-        this.client.user.setPresence({ activity: atividade });
-
-        const allShardsPing = await this.client.shard.broadcastEval('this.ws.ping');
-        const allShardsUptime = await this.client.shard.broadcastEval('this.ws.client.uptime');
-        const guildsPerShardCount = await this.client.shard.broadcastEval('this.guilds.cache.size');
-
-        allShardsPing.map(async (shardPing, id) => {
-          this.client.repositories.statusRepository.CreateOrUpdate(
-            id,
-            shardPing,
-            Date.now(),
-            guildsPerShardCount[id],
-            allShardsUptime[id],
-          );
-        });
-      }, 1000 * 60);
-    }
+    if (!this.client.user) return;
+    if (!this.client.shard) return;
+
+    const INTERVAL = 1000 * 60;
+    const MENHERA_DEV_ID = '708014856711962654';
+    const FIRST_SHARD_ID = 0;
+
+    const updateActivity = async (shard: number) => {
+      if (!this.client.user) return;
+
+      const activity = await http.getActivity(shard);
+      return this.client.user.setPresence({ activity });
+    };
+
+    const saveCurrentBotStatus = async () => {
+      if (!this.client.shard) return;
+
+      const allShardsPing = await this.client.shard.broadcastEval('this.ws.ping');
+      const allShardsUptime = await this.client.shard.broadcastEval('this.ws.client.uptime');
+      const guildsPerShardCount = await this.client.shard.broadcastEval('this.guilds.cache.size');
```

Comment:
```suggestion
      const allShardsPing = await this.client.shard.fetchClientValue('ws.ping');
      const allShardsUptime = await this.client.shard.fetchClientValue('ws.client.uptime');
      const guildsPerShardCount = await this.client.shard.fetchClientValue('guilds.cache.size');
```

Reply:
Doesn't that return just the value of the current shard?

Reply:
No, it fetches the values from every shard

Reply:
oo nice nice